src/asphalt/core/cli.py

```diff
@@ -67,6 +81,27 @@ def run(configfile, unsafe: bool, loop: Optional[str], service: Optional[str]) -
         ), "the document root element must be a dictionary"
         config = merge_config(config, config_data)
 
+    # Read the configuration from the CLI
+    for cli_conf in set_:
+        if "=" not in cli_conf:
+            click.echo(f"Configuration must be set with '=', got: {cli_conf}")
+            raise click.Abort()
+
+        key, value = cli_conf.split("=", 1)
+        keys = [k.replace("\\.", ".") for k in re.split("(?<!\\\\)\\.", key)]
+        if len(keys) > 1:
+            # subcomponent configuration
+            last_i = len(keys) - 1
+            d = config.setdefault("component", {}).setdefault("components", {})
+            for i, k in enumerate(keys):
+                if i == last_i:
+                    d[k] = value
+                else:
+                    d = d.setdefault(k, {})
+        else:
+            # root component configuration
+            config["component"][key] = value
```

Comment:
Can you explain this part to me? How would I set `max_threads` with this system?

Reply:
Hmmm you're right, it currently only works for component configuration. I'll rework it so that it's generic, to allow setting anything that can be set through YAML.

Reply:
Fixed in https://github.com/asphalt-framework/asphalt/pull/50/commits/cea126c29631edde6bc497ce184f036922c9330a.